src/cocoa/toga_cocoa/widgets/base.py

```diff
@@ -24,17 +28,20 @@ def container(self):
 
     @container.setter
     def container(self, container):
-        self._container = container
-        self._container.native.addSubview(self.native)
-        self.constraints.container = container
+
+        if container is None:
+            self.constraints.container = container
+            self._container = container
```

Comment:
By way of being explicit, it would probably be better to set 
```
self.constraints.container = None
self._container = None
```

The end effect is exactly the same (since `container` must be `None`), but it reads as an explicit 'set to None'.

Reply:
Done.